java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -155,6 +151,27 @@ protected void copyFromRunfiles(String path, String destpath) throws IOException
     Files.copy(origin.toPath(), dest.toPath());
   }
 
+  /**
+   * Copy the whole directory from the runfiles under {@code rootDirectoryPath} to the current workspace.
+   */
+  protected void copyFromRunfilesDirectory(final String rootDirectoryPath) throws IOException {
+    File root = getRunfile(rootDirectoryPath);
+    try (Stream<Path> paths = Files.walk(Paths.get(root.toURI()))) {
+      paths.filter(path -> new File(path.toUri()).isFile())
+              .forEach(file -> {
+                String relativeToRunfiles = file.toAbsolutePath().toString().substring(runfileDirectory.getPath().length());
```

Comment:
can you take a look at `Path#relativize`? I don't think it's what you need but it's definitely a neighbor :)
something with `file` and `runFilesDirectory`

Reply:
Actually, it's exactly `relativize`